TypeSystem/flow-analysis/type_of_interest_A03_t03.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that at type `T` is a type of interest for a variable `x`
+/// in a set of tested types `tested` if `tested` contains a type `S` such that
+/// `T` is `S`, or `T` is `NonNull(S)`.
+///
+/// @description Checks that a type `Object` cannot be made a type of interest
+/// for a variable.
```

Comment:
I don't see a reason why this should not be possible. There is nothing special about `Object` in this respect.

Of course, the declared type must be a supertype of `Object` (that is, a top type) in order to allow the promotion to `Object` to take place.